ksqldb-execution/src/main/java/io/confluent/ksql/execution/codegen/SqlToJavaVisitor.java

```diff
@@ -1003,13 +1005,13 @@ private CastVisitor() {
     }
 
     private static Pair<String, SqlType> castDecimal(
-        final Pair<String, SqlType> expr, final SqlType sqltype, final SqlType returnType
+        final Pair<String, SqlType> expr, final SqlType returnType
     ) {
-      if (!(sqltype instanceof SqlDecimal)) {
-        throw new KsqlException("Expected decimal type: " + sqltype);
+      if (!(returnType instanceof SqlDecimal)) {
+        throw new KsqlException("Expected decimal type: " + returnType);
```

Comment:
note: these methods were always called with the same param for `returnType` and `sqlType`: removed the duplication.